conans/test/integration/options_test.py

```diff
@@ -64,3 +64,70 @@ class ConanLib(ConanFile):
                       client.user_io.out)
         conaninfo = load(os.path.join(client.current_folder, CONANINFO))
         self.assertNotIn("zlib:shared=True", conaninfo)
+
+    def default_options_test(self):
+        client = TestClient()
+        conanfile = """
+from conans import ConanFile
+
+class MyConanFile(ConanFile):
+    name = "MyConanFile"
+    version = "1.0"
+    options = {"config": %s}
+    default_options = "config=%s"
+
+    def configure(self):
+        if self.options.config:
+            self.output.info("Boolean evaluation")
+        if self.options.config is None:
```

Comment:
``if self.options.config == None`` should work, please test it.

Reply:
This is not working because of this line here: https://github.com/conan-io/conan/blob/5aa4cfd55c89dd52bf8fce37d3115c7c0b572653/conans/model/options.py#L327

However behavior could be change as I think this is not documented.